### DISCUSSION TITLE
Supabase Inner join Query

### DISCUSSION THREAD

**Discussion by kouji-dev** [supabase/supabase#3816] [Questions] [answered]
Hello There,

is there any way to make the results nicer ? 

`[
  {
    "description": "I'm looking for a diamond 3 jungler",
    "title": "Ranked Flex Climb",
    "by": null,
    "tags": [
      {
        "tag": 1,
        "post": 1,
        "tags": {
          "label": "Tryhard",
          "color": "B91646"
        }
      },
      {
        "tag": 2,
        "post": 1,
        "tags": {
          "label": "duo",
          "color": "A2D2FF"
        }
      }
    ]
  }
]`

this is the query I'm using.

![image](https://user-images.githubusercontent.com/22368199/140613852-07ceb5b6-c181-4b23-af9d-ca557683b9a2.png)

My schema is defined as follow:

Post has 0...n Tag
Tag has 0...n Post

So i defined an association table called Post_Tag where I have only ids Post_Tag(post, tag)
and Post(id, description, title), Tag(id, label, color, icon)

Now i want to query the posts and join them with assigned tags in order to display them but the result is not nice i get back the Tag object inside the Post_Tag object, is there any way i can get the Post object with an attribute that has tags from Table Tag ?

**Comment by steve-chavez** [comment, marked as answer]
If there's a many-to-many relationship between Post and Tag you can request it directly without specifying the association table:
```js
const { data, error } = await supabase
  .from('Post')
  .select(`
    description, 
    title,
    by,
    tags:Tag(
      label,
      color
    )
  `)
```

**Reply by soedirgo** [reply]
In this case the result should be:
```
[
  {
    "description": "I'm looking for a diamond 3 jungler",
    "title": "Ranked Flex Climb",
    "by": null,
    "tags": [
      {
        "label": "Tryhard",
        "color": "B91646"
      },
      {
        "label": "duo",
        "color": "A2D2FF"
      }
    ]
  }
]
```
which looks like what you'd expect.

**Reply by NixBiks** [reply]
how would you do this in typescript?
